tracing: include zipkin sampled & debug headers in log output

risicle opened the pull request:
If I'm going to be going to the effort of bumping utils out to everything, may as well include these.

This includes the values of the "sampled" and "debug" zipkin flags in log output. Please only take issue with the json key names `tracingIsSampled` & `tracingDebugFlag` if you have a significantly better suggestion as this is something we could discuss forever...

samuelhwilliams commented:
Some of our logs in Kibana already include these details under the keys `debug_flag` and `is_sampled`. Not sure how they get pulled in, and/or why we need this to log them separately, but we should make sure we aren't duplicating this under different keys.

risicle commented:
Oh. I don't think that's deliberate. Hmm...

risicle commented:
I was under the impression that if a field weren't explicitly mentioned in `LOG_FORMAT` or `LOG_FORMAT_EXTRA_JSON_KEYS` it wouldn't get output...

katstevens commented on dmutils/logging.py:
```diff
@@ -148,6 +150,8 @@ def process_log_record(self, log_record):
             ("parent_span_id", "parentSpanId",),
             ("app_name", "application",),
+            ("tracing_is_sampled", "tracingIsSampled",),
+            ("tracing_debug_flag", "tracingDebugFlag",),
```
I think there's a test for these field renamings that we can add these too?

risicle replied:
There are tests but are we going to add a test for every single pair we add here?

risicle replied:
Oh I think I see the bit you mean...

katstevens replied:
I'm hoping we're not going to add too many more!

risicle commented:
Yup. Our logs do indeed spit out the raw unexpected bits of log context and this slipped through due to this module's preference for unit testing rather than "does it actually work?" testing. More tomorrow.

risicle commented:
I've pushed a new version which does a couple of extra things: adds a high level test of some logging behaviour because it was lacking clarity over what was *supposed* to happen in various cases involving "required" keys and it also failed to pick up the fact that the "missing key" functionality was injecting spurious keys into the final json output as a side effect of its dictionary-stuffing-with-`{foo: missing key}` functionality.

I also fixed that and switched the "missing keys" code's innards around a bit based on some feedback a while ago from @katstevens. See that commit's comment for more detail.

risicle commented:
Oh and I also switched the output key names to remove the `tracing` prefix as the previous decision seems to have gone that way...
